shared/contracts/BaseTemplate.sol

```diff
@@ -56,155 +62,178 @@ contract BaseTemplate is APMNamehash, IsContract {
         dao = daoFactory.newDAO(this);
         emit DeployDao(address(dao));
         acl = ACL(dao.acl());
-        _createPermission(acl, dao, dao.APP_MANAGER_ROLE());
+        _createPermissionForTemplate(acl, dao, dao.APP_MANAGER_ROLE());
```

Comment:
It may be worth adding a comment saying that the template now has both the `kernel.APP_MANAGER_ROLE` and `acl.CREATE_PERMISSIONS_ROLE`, and that at the end they should use `_transferRootPermissionsFromTemplate()`